src/seedu/addressbook/data/person/Address.java

```diff
@@ -23,10 +29,26 @@
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
         this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+        UpdateAddressComponents(trimmedAddress);
+        String reconstructedAddress = ReconstructAddressFromComponents();
+        if (!isValidAddress(reconstructedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        this.value = reconstructedAddress;
+    }
+
+    private String ReconstructAddressFromComponents() {
+        return this.blk.toString() + SEPARATOR_COMMA_SPACE +
+                    this.street.toString() + SEPARATOR_COMMA_SPACE +
+                    this.unit.toString() + SEPARATOR_COMMA_SPACE +
+                    this.pc.toString();
+    }
+
+    private void UpdateAddressComponents(String trimmedAddress) throws IllegalValueException {
+        this.blk = new Block(trimmedAddress.split(",")[0]);
```

Comment:
Would it be better to store trimmedAddress.split(",") in a separate variable? It needs to be done just once.